tests/unit-tests/test-subscription-integration.php

```diff
@@ -0,0 +1,432 @@
+						],
+					],
+				],
+				3  => [
+					'id'         => 3,
+					'name'       => 'Plan 3B',
+					'network_id' => '',
+				],
+			],
+		];
+
+		update_option( Membership_Plan_Updated::OPTION_NAME, $plans );
+
+		self::$user_with_sub_on_1 = wp_insert_user(
+			array(
+				'user_login' => 'user_with_sub_on_1',
+				'user_pass'  => '123',
+				'role'       => 'subscriber',
+			)
+		);
+
+		self::$user_with_sub_on_2 = wp_insert_user(
+			array(
+				'user_login' => 'user_with_sub_on_2',
+				'user_pass'  => '123',
+				'role'       => 'subscriber',
+			)
+		);
+
+		self::$user_with_cancelled_sub_on_2 = wp_insert_user(
+			array(
+				'user_login' => 'user_with_cancelled_sub_on_2',
+				'user_pass'  => '123',
+				'role'       => 'subscriber',
+			)
+		);
+
+		self::$user_with_non_network_sub_on_1 = wp_insert_user(
+			array(
+				'user_login' => 'user_with_non_network_sub_on_1',
+				'user_pass'  => '123',
+				'role'       => 'subscriber',
+			)
+		);
+
+		self::$user_without_subs = wp_insert_user(
+			array(
+				'user_login' => 'user_without_subs',
+				'user_pass'  => '123',
+				'role'       => 'subscriber',
+			)
+		);
+
+		add_user_meta(
+			self::$user_with_sub_on_1,
+			Subscription_Changed::USER_SUBSCRIPTIONS_META_KEY,
+			[
+				'http://site1' => [
+					500 => [
+						'id'       => 500,
+						'name'     => 'Subscription 500',
+						'status'   => 'active',
+						'products' => [
+							100 => [
+								'id'   => 100,
+								'name' => 'Plan 100',
+							],
+						],
+					],
+				],
+			]
+		);
+
+		add_user_meta(
+			self::$user_with_sub_on_2,
+			Subscription_Changed::USER_SUBSCRIPTIONS_META_KEY,
+			[
+				'http://site2' => [
+					501 => [
+						'id'       => 501,
+						'name'     => 'Subscription 501',
+						'status'   => 'pending-cancel',
+						'products' => [
+							220 => [
+								'id'   => 220,
+								'name' => 'Plan 220',
+							],
+						],
+					],
+				],
+			]
+		);
+
+		add_user_meta(
+			self::$user_with_cancelled_sub_on_2,
+			Subscription_Changed::USER_SUBSCRIPTIONS_META_KEY,
+			[
+				'http://site2' => [
+					502 => [
+						'id'       => 502,
+						'name'     => 'Subscription 502',
+						'status'   => 'cancelled',
+						'products' => [
+							220 => [
+								'id'   => 220,
+								'name' => 'Plan 220',
+							],
+						],
+					],
+				],
+			]
+		);
+
+		add_user_meta(
+			self::$user_with_non_network_sub_on_1,
+			Subscription_Changed::USER_SUBSCRIPTIONS_META_KEY,
+			[
+				'http://site1' => [
+					503 => [
+						'id'       => 503,
+						'name'     => 'Subscription 503',
+						'status'   => 'active',
+						'products' => [
+							999 => [
+								'id'   => 999,
+								'name' => 'Plan 999',
+							],
+						],
+					],
+				],
+			]
+		);
+	}
+
+	/**
+	 * Test get_network_membership_plans
+	 */
+	public function test_get_network_membership_plans() {
+		$method = self::get_method( 'get_network_membership_plans' );
+
+		$result = $method->invoke( null, 'net-1' );
```

Comment:
It'd be more readable if the variable names were more descriptive, e.g. `$network_membership_plans` instead of `$result`, and `$get_network_membership_plans_method` instead of `$method`.

Reply:
https://github.com/Automattic/newspack-network/pull/135/commits/baf897d2e7e50d5e57dcfc81714c26fb57b9f868